apps/libexec/merlinkey.py.in

```diff
@@ -30,4 +32,10 @@ def cmd_generate(args):
 		print 'Key generation failed'
 		sys.exit(1)
 	else:
+		uid = pwd.getpwnam("monitor").pw_uid
+		gid = grp.getgrnam("apache").gr_gid
+		os.chmod(path + "/key.priv", 0400)
```

Comment:
Even better might be to set `os.umask()` before creating the keys, to avoid leaving it readable for a short while, or if the program crashes right after creating the keys.

Reply:
I do not think that will work, as the keys are generated by: https://github.com/ITRS-Group/monitor-merlin/blob/feature/MON-12109-encrypted-merlin/tools/keygen.c

Reply:
You could perhaps do what this stackoverflow suggests: https://stackoverflow.com/questions/36185877/how-can-i-set-the-umask-of-a-process-i-am-spawning

Reply:
I think child processes inherit umask, so if umask is set in the Python process before calling the subprocess, then the spawned subprocess should get that umask as well.

Reply:
Ok, I set the umask, but kept the changing of ownership as before. This mostly because I think it is very error prone running it with `asmonitor` from the start, as the monitor user may not be able to create files at all locations (but as long as the monitor user can read that location and the file within it is owned by monitor I think that is OK).

Reply:
Sounds good.